Backports for Julia 1.4.1

KristofferC opened the pull request:
Backported PRs:
- [x] JuliaLang/julia#34760 - Add AddrSpacePtr type
- [x] JuliaLang/julia#34906 - Fix regression in inlining of invoke
- [x] JuliaLang/julia#34872 - Fix inverse of SVD of complex matrix (#34866)
- [x] JuliaLang/julia#34986 - Sockets: fix return value of getpeername/getsockname
- [x] JuliaLang/julia#34989 - Rebase of JuliaLang/julia#28526 on master: Sparse cholesky solve with sparse RHS
- [x] JuliaLang/julia#35006 - walkdir: avoid symlink loops when `follow_symlinks == false`
- [x] JuliaLang/julia#35064 - Support git credential helper resetting
- [x] JuliaLang/julia#35136 - fix bug in t-function for `>:`
- [x] JuliaLang/julia#35144 - speed up llvmcall unique name generation
- [x] JuliaLang/julia#35184 - add compat annotation for map!(f, values(dict)) (from JuliaLang/julia#31223)
- [x] JuliaLang/julia#35192 - Remove outdated paragraph from deepcopy docstring
- [x] JuliaLang/julia#35202 - fix JuliaLang/julia#35201, syntax error with named splatting inside function call
- [x] JuliaLang/julia#35227 - Bump Documenter to 0.24.7
- [x] JuliaLang/julia#35244 - fix JuliaLang/julia#35225, issubset on empty ranges
- [x] JuliaLang/julia#35180 - Linux: Remove `--nodetach` option for `xsel` copy.
- [x] JuliaLang/julia#35310 - fix JuliaLang/julia#35295, typo in change to `clipboard` on linux
- [x] JuliaLang/julia#35356 - Update `UnicodeData` checksum
- [x] JuliaLang/julia#35298 - Fix offset when widening collections
- [x] JuliaLang/julia#35315 - fix method errors printing for missing
- [x] JuliaLang/julia#35309 - fix JuliaLang/julia#35305, need escaping when printing string macro calls
- [x] JuliaLang/julia#35239 - fix a bug preventing inlining of getindex of const globals
- [x] JuliaLang/julia#35300 - [LLVM8] add patches for D75072 and D65174
- [x] JuliaLang/julia#35394 - fix JuliaLang/julia#35391, macro hygiene bug in gc_preserve
- [x] JuliaLang/julia#35387 - [LateGCLowering] Fix skipped Select lifting
- [x] JuliaLang/julia#35435 - [backport JuliaLang/julia#35274] fix JuliaLang/julia#35272, searchsorted(3:-1:1, 2.5, rev=true)

Need manual backport:
- [ ] JuliaLang/julia#34935 - [WIP] Bump SuiteSparse to 5.4.0+6






Non-merged PRs with backport label:
- [ ] JuliaLang/julia#35232 - LibGit2: add resolve_url to RemoteCallbacksStruct for LibGit2 0.99.0

mgkuhn committed:
adding escape_raw_string()

Implementation of the raw-string escaping convention. Can also be
used to escape command-line arguments for Windows C/C++/Julia
applications, which use the same \" escaping convention as
Julia non-standard string literals.

(cherry picked from commit cdc5e791888e61b8f78ee9ccb2fcd21b94c7493e)

thomasfaingnaert committed:
Add AddrSpacePtr type

(cherry picked from commit 8cb8f0c485a81082813dde82b7caba29e2343c36)

Keno committed:
Fix regression in inlining of invoke (#34906)

When we added the check to prevent inlining through ambiguous methods,
we failed exclude this check for calls to `invoke` (which skip ambiguous
methods, since the method is explicitly selected). Fixes #34900.

(cherry picked from commit 6abc8521bfe3af422866cb91b196367706d239b6)

carstenbauer committed:
Fix inverse of SVD of complex matrix (#34872)

Closes #34866

(cherry picked from commit f7093312ca8747765c96b8571f47b4bfa2e53708)

Keno committed:
Sockets: fix return value of getpeername/getsockname (#34986)


(cherry picked from commit 598209de2161df196419cb39a2d595c6abf8763b)

ViralBShah committed:
Rebase of #28526 on master (#34989)

* fix \ SparseVector

* split (\) for sparsevector;
test for Factor and Symmetric Sparse Matrix

Co-authored-by: Chi Po Choi <choi.chi.po@gmail.com>
(cherry picked from commit d2f967766ca1622e244fdb8136c2f5374d697fee)

staticfloat committed:
walkdir: avoid symlink loops when `follow_symlinks == false` (#35006)

* walkdir: avoid symlink loops when `follow_symlinks == false`

Because `isdir()` attempts to dereference symlinks, attempting to
`walkdir()` trees that contain symlink loops errors out. This change
modifies `walkdir()` to treat all symlinks as files when
`follow_symlinks == false`.

* rm: When checking `filemode()`, use `lstat()` to avoid following symlinks

(cherry picked from commit 178ac974b5e9f649c0ffb3ed78752dc9241b7a5a)

omus committed:
Support git config credential resetting (#35064)


(cherry picked from commit 0f1b1192735e1c05c5aa0eab85bef92250abe05c)

JeffBezanson committed:
fix bug in t-function for `>:` (#35136)

(cherry picked from commit 2a5bb59facdce6c8d987a3752f2b50e2e06d7bef)

JeffBezanson committed:
speed up llvmcall unique name generation (#35144)


(cherry picked from commit 5d5f799ed43b3ef458e16fba4f34533f5719c614)

rfourquet committed:
add compat annotation for map!(f, values(dict)) (from #31223) (#35184)


(cherry picked from commit 644753491de0d5d933c96e6f1f5876b3c7603cf4)

cstjean committed:
Remove outdated paragraph from deepcopy docstring (#35192)


(cherry picked from commit 6baa701c571ff3dd976e45fb9892d5430993edb5)

JeffBezanson committed:
fix #35201, syntax error with named splatting inside function call (#35202)

(cherry picked from commit 59d7522eaecc86783078eac8746516e3a0d8a308)

mortenpi committed:
Bump Documenter to 0.24.7 (#35227)

(cherry picked from commit d37778535283b5484ec76759d227612190a3b63d)

maleadt commented:
Could https://github.com/JuliaLang/julia/pull/34760 be considered? It is technically a new feature, but it's unexported, completely unused by Base, but greatly simplifies quite some code in CUDAnative. Should be a clean backport too.

fredrikekre commented on base/gcutils.jl:
```diff
@@ -83,11 +83,50 @@ enable(on::Bool) = ccall(:jl_gc_enable, Int32, (Int32,), on) != 0
+julia> let
+           x = "Hello"
+           p = pointer(x)
+           GC.@preserve x @ccall strlen(p::Cstring)::Cint
```
`@ccall` is not  in 1.4.

KristofferC replied:
Oops, let's drop that commit then. Good catch.

rfourquet committed:
fix #35225, issubset on empty ranges (#35244)


(cherry picked from commit 64ee8ab993db6ed806ca06c950dc6475f2e7dc0c)

JeffBezanson committed:
fix a bug preventing inlining of getindex of const globals (#35239)

(cherry picked from commit 1f787492c3ac44797941f10c8bd452a1a7eef0f8)

nalimilan committed:
change libLLVM suffix to match build-configuration

Backport of 5e0cab69966eb989f380492dda7590bca3a3ba8d.

timholy committed:
Fix offset when widening collections

(cherry picked from commit 1da24809378ab7c849f59c09270769e24d15a7b4)

vchuravy committed:
[LLVM8] add patches for D75072 and D65174

(cherry picked from commit d20184f90e8ddacb40222154714bc6140eaaa4a3)

vchuravy committed:
yank LLVM assert checksums

JeffBezanson committed:
fix #35305, need escaping when printing string macro calls

(cherry picked from commit b6dd448e3afab4a4a0015bb2e6122b97f649e5da)

Commit:
Linux: Remove `--nodetach` option for `xsel` copy. (#35180)

The `--nodetach` option causes `clipboard(x)` to hang indefinitely on (most? All which I have tried) linux systems. Removing this option fixes `clipboard(x)`, but briefly spawns a child process. Since `clipboard(x)` is designed for interactive use, this shouldn't be an issue.

(cherry picked from commit 5dd66767f7884fd7de104d9b8144841774ecbdf0)

vchuravy committed:
bump LLVM BB release

KristofferC committed:
fix method error printing for missing (#35315)

fix method error printing for missing

(cherry picked from commit 0b0b126ce73defed660aab11942d9307915f1926)

JeffBezanson committed:
fix #35295, typo in change to `clipboard` on linux (#35310)


(cherry picked from commit 33df293a6a46d83bdd3a9caa08b311ef74452d60)

staticfloat committed:
Update `UnicodeData` checksum (#35356)

Upstream has apparently modified this file as of March 30th 2020, which
is annoying as it has broken all older builds of Julia attempting to
download and check this file.

(cherry picked from commit c9805191e32fde9d4aa55e59493178f47444926e)

KristofferC commented:
@staticfloat, do you need to clear out some cache on the buildbot for the checksum of the unicode file to update (see CI errors).

vchuravy committed:
Merge pull request #35313 from JuliaLang/vc/bp_llvm8

[LLVM8] patch maintenance

staticfloat commented:
Weird, I cleared this out.  I wonder how they got the old file again?  That doesn't seem right.

You can clear them yourself by running the `nuke` jobs on buildbot, e.g. go here and click the `nuke` button in the top right: https://build.julialang.org/#/builders/139. It will queue a cleanup job that will run on every buildbot, and then you can trigger a build after that cleanup by just re-running the failing CI jobs.

KristofferC commented:
I see that you ran the nuke build and I restarted the workers but doesn't seem to help.

staticfloat commented:
Ah, I see the problem.  We have two branches that are downloading two different versions of UnicodeData with the same name.  How did this ever work?

staticfloat committed:
Version `UnicodeData.txt` filename (#35383)

We've been downloading multiple different versions of `UnicodeData.txt`
across different branches and trying to keep a consistent checksum.

(cherry picked from commit 922891ccbfcd74a0d10eca4f197b976ec08c20fd)

KristofferC commented:
@nanosoldier `runbenchmarks(ALL, vs = ":release-1.4")`

KristofferC commented:
@nanosoldier `runtests(ALL, vs = ":release-1.4")`

vchuravy committed:
bump LLVM release

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-610287648) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/029bcf2b4aa50c5f262521292612e901f05ee270/571c4c4_vs_b8e9a9e/report.md). cc @ararslan

KristofferC commented:
@nanosoldier `runtests(ALL, vs = ":release-1.4")`

nanosoldier commented:
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-610822494):
```
NanosoldierError: failed to run tests: AssertionError: Cannot compare identical Julia builds
```
Logs and partial data can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/5a20f219087cd591cc3204deee719e64e51a5e5d/pkgeval/by_hash/c53191a_vs_b8e9a9e/report.md)
cc @maleadt

maleadt commented:
Not sure what went wrong here, so I added some debug logging.
@nanosoldier `runtests(ALL, vs = ":release-1.4")`

nanosoldier commented:
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-610894347):
```
NanosoldierError: failed to run tests: UndefVarError: cfg not defined
```
Logs and partial data can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/0b83b2773439822acb1a75d14da75eaf88d6c12e/pkgeval/by_hash/c53191a_vs_b8e9a9e/report.md)
cc @maleadt

maleadt commented:
@nanosoldier `runtests(ALL, vs = ":release-1.4")`

nanosoldier commented:
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-610896217):
```
NanosoldierError: failed to run tests: AssertionError: Cannot compare identical Julia builds
```
Logs and partial data can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/f9e89d190be563ef6917a7614cc6d4973304cb5b/pkgeval/by_hash/c53191a_vs_b8e9a9e/report.md)
cc @maleadt

maleadt commented:
I think I know what's happening:
@nanosoldier `runtests(ALL, vs = ":release-1.4")`

JeffBezanson committed:
fix #35391, macro hygiene bug in gc_preserve (#35394)


(cherry picked from commit 8256be444d4420b485893993cd3f391be397ad1b)

Keno committed:
[LateGCLowering] Fix skipped Select lifting (#35387)

Fixes #35341

(cherry picked from commit 7a4ea219b5e4a2df3dc354476127e06d18b5506d)

vchuravy committed:
bump OpenBLAS release

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-610901395) has completed - possible new issues were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/824edc01bd2e9b62e669b7062006196fa793c342/pkgeval/by_hash/c53191a_vs_b8e9a9e/report.md). cc @maleadt

KristofferC commented:
@nanosoldier `runtests(["AugmentedGaussianProcesses", "CombineML", "ControlSystemIdentification", "DiffEqGPU", "EfficientGlobalOptimization", "FunctionOperators", "ImageDistances", "InfiniteLinearAlgebra", "InterProcessCommunication", "SpatialJackknife", "Unicode_Entities"], vs = ":release-1.4")`

KristofferC committed:
Merge pull request #35405 from JuliaLang/vc/bump_bb_1.4

[1.4.1] Update LLVM and OpenBLAS BB release

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-611357306) has completed - possible new issues were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/591e1aaf0dad689371d2d0c7400e94350525774e/pkgeval/by_hash/c53191a_vs_b8e9a9e/report.md). cc @maleadt

KristofferC commented:
We bumped LLVM and OpenBLAS so better rerun all the checks...

@nanosoldier `runbenchmarks(ALL, vs = ":release-1.4")`

KristofferC commented:
@nanosoldier `runtests(ALL, vs = ":release-1.4")`

rfourquet commented:
When is planned the release of 1.4.1? I will try before then to make a 1.4 version of #35274 which fixes a bug.

KristofferC commented:
> When is planned the release of 1.4.1? 

As soon as this branch looks ready.,

nanosoldier commented:
[Your benchmark job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-611968111) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/a395a55cf78e43cad56adda77020f75d03401267/3fd5790_vs_b8e9a9e/report.md). cc @ararslan

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-611968237) has completed - possible new issues were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/2e37f82c612a869c8f15976e33dbe0a011117e05/pkgeval/by_hash/44fa722_vs_b8e9a9e/report.md). cc @maleadt

KristofferC commented:
@nanosoldier `runtests(["ApproxBayes", "Autologistic", "BlobTracking", "CombineML", "DiffEqGPU", "HELICS", "MAT", "MusicXML", "OhMyREPL"], vs = ":release-1.4")`

nanosoldier commented:
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/35229#issuecomment-612579052) has completed - no new issues were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/165fc96caa399b45b3a32be21bb40c2c86cff75a/pkgeval/by_hash/44fa722_vs_b8e9a9e/report.md). cc @maleadt

rfourquet committed:
[backport #35274] fix #35272, searchsorted(3:-1:1, 2.5, rev=true) (#35435)

maleadt commented:
The LLVM asserts build is broken on this branch (works with v1.4.0):

```
curl: (22) The requested URL returned error: 404 Not Found
make[1]: *** [/home/tim/Julia/julia/deps/llvm.mk:579: /home/tim/Julia/julia/deps/srccache/LLVM_full_assert.v8.0.1-3.x86_64-linux-gnu-cxx11.tar.gz] Error 22
```

https://github.com/JuliaBinaryWrappers/LLVM_full_assert_jll.jl/releases/download/LLVM_full_assert-v8.0.1+3/LLVM_full_assert.v8.0.1.x86_64-linux-gnu-cxx11.tar.gz doesn't exist whereas https://github.com/JuliaBinaryWrappers/LLVM_full_assert_jll.jl/releases/download/LLVM_full_assert-v8.0.1+0/LLVM_full_assert.v8.0.1.x86_64-linux-gnu-cxx11.tar.gz does. cc @vchuravy
